PurrSoft/PurrSoft.Api/Controllers/AnimalFosterMapController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Common;
+using PurrSoft.Application.Models;
+using PurrSoft.Application.Queries.AnimalFosterMapQueries;
+using PurrSoft.Application.QueryOverviews;
+using System.Net;
+using static PurrSoft.Application.Commands.AnimalFosterMapCommands.AnimalFosterMapCommands;
+
+namespace PurrSoft.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class AnimalFosterMapController : BaseController
+{
+	[HttpGet()]
+	[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
+	[ProducesResponseType(typeof(CollectionResponse<AnimalFosterMapDto>), (int)HttpStatusCode.OK)]
+	public async Task<CollectionResponse<AnimalFosterMapOverview>> GetAnimalFosterMaps([FromQuery] GetFilteredAnimalFosterMapsQueries query)
+	{
+		return await Mediator.Send(query, new CancellationToken());
+	}
+
+	[HttpGet("{id}")]
+	[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
+	[ProducesResponseType(typeof(AnimalFosterMapDto), (int)HttpStatusCode.OK)]
+	[ProducesResponseType((int)HttpStatusCode.BadRequest)]
+	[ProducesResponseType((int)HttpStatusCode.NotFound)]
+	public async Task<IActionResult> GetAnimalFosterMapById(string id)
+	{
+		try
+		{
+			AnimalFosterMapDto? animalFosterMapDto = await Mediator.Send(new GetAnimalFosterMapById { Id = id }, new CancellationToken());
+
+			if (animalFosterMapDto == null)
+			{
+				return NotFound();
+			}
+
+			return Ok(animalFosterMapDto);
+		}
+		catch (ValidationException ex)
+		{
+			return BadRequest(new CommandResponse(ex.Errors.ToList()));
+		}
+	}
+
+	[HttpPost()]
+	[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
+	[ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.OK)]
+	[ProducesResponseType((int)HttpStatusCode.BadRequest)]
```

Comment:
Here too

Reply:
Is something like this fine?

`[HttpPost()]
[Authorize(AuthenticationSchemes = "Bearer", Roles = "Manager, Foster")]
[ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.OK)]
[ProducesResponseType(typeof(CommandResponse), (int)HttpStatusCode.BadRequest)]`